server/stt/stt_info_porssi.py

```diff
@@ -0,0 +1,213 @@
+                ),
+            },
+        }
+
+    def can_parse(self, xml):
+        return (
+            xml.tag == "{http://iptc.org/std/NewsML/2003-10-10/}NewsML"
+            and xml.get("Version", "") == "1.2"
+        )
+
+    def parse(self, xml, provider=None):
+        try:
+            # Remove namespace for easier XPath
+            xml = deepcopy(xml)
+            for elt in xml.iter():
+                elt.tag = elt.tag.replace("{" + NEWSML_NS + "}", "")
+            news_items = xml.findall("NewsItem")
+
+            selected = None
+            for news_item in news_items:
+                try:
+                    lang = news_item.xpath(
+                        "NewsComponent/DescriptiveMetadata/Language/@FormalName"
+                    )[0]
+                except IndexError:
+                    public_id = (
+                        news_item.findtext(
+                            "Identification/NewsIdentifier/PublicIdentifier"
+                        )
+                        or news_item.findtext(
+                            "Identification/NewsIdentifier/NewsItemId"
+                        )
+                        or ""
+                    )
+                    logger.warning("Missing language in item; public_id=%s", public_id)
+                    continue
+
+                if selected is None or lang in ("fi", "sv", "en"):
+                    selected = news_item
+                if lang == "fi":
+                    break
+
+            if selected is None:
+                raise ParserError.parseFileError(
+                    source=etree.tostring(xml, encoding="unicode")
+                )
+
+            # Determine language from the selected item (fallback to fi)
+            language_elements = selected.xpath(
+                "NewsComponent/DescriptiveMetadata/Language/@FormalName"
+            )
+            sel_lang = language_elements[0] if language_elements else "fi"
+
+            # Compute versioncreated; prefer ThisRevisionCreated; ensure tz-aware
+            rev_created = selected.findtext("NewsManagement/ThisRevisionCreated")
+            if rev_created:
+                dt = dtparse(rev_created)
+                if dt.tzinfo is None:
+                    dt = dt.replace(tzinfo=timezone.utc)
+                versioncreated = dt
+            else:
+                versioncreated = datetime.utcnow().replace(tzinfo=timezone.utc)
+
+            # Compute GUID as a URN based on PublicIdentifier or NewsItemId (consistent format)
+            public_id = selected.findtext(
+                "Identification/NewsIdentifier/PublicIdentifier"
+            )
+            news_item_id = selected.findtext("Identification/NewsIdentifier/NewsItemId")
+            if public_id:
+                guid_value = f"urn:stt:info-porssi:{public_id}"
+            elif news_item_id:
+                guid_value = f"urn:stt:info-porssi:{news_item_id}"
+            else:
+                guid_value = (
```

Comment:
seems like it's checking before that there is an id so this won't be ever used if I got it right, would be good to remove it then